### PR TITLE
Gh 3928 produce commitments

### PR DESCRIPTION
**Pull request by velzevur**

### CODE REVIEW

File: apps/aecore/src/aec_parent_connector.erl [hunk 1]
```diff
@@ -67,15 +78,19 @@ start_link() ->
                     password => "Pass"
                     }],
     ParentConnMod = aehttpc_aeternity,
-    start_link(ParentConnMod, FetchInterval, ParentHosts).
+    start_link(ParentConnMod, FetchInterval, ParentHosts, <<"local_testnet">>,
```
Comment (velzevur) [inline apps/aecore/src/aec_parent_connector.erl:81]: Note to myself: drop the hardcoded network id....

**Review by uwiger — APPROVED** [review]

File: apps/aecore/src/aec_consensus_hc.erl [hunk 2]
```diff
@@ -82,15 +87,26 @@ assert_config(_Config) -> ok.
...
+                #{  <<"fetch_interval">> := FetchInterval,
+                    <<"nodes">> := Nodes0
+                 } = Polling
+          }} = Config,
+    CacheSize = maps:get(<<"cache_size">>, Polling, 200),
     Stakers =
         lists:map(
-            fun(#{<<"pub">> := EncodedPubkey, <<"priv">> := EncodedPrivkey}) ->
+            fun(#{<<"hyper_chain_account">> := #{<<"pub">> := EncodedPubkey,
+                                                 <<"priv">> := EncodedPrivkey},
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aecore/src/aec_consensus_hc.erl:106]: This is not the subject of this PR. Keys management is a broader subject tracked as [#3937](https://github.com/aeternity/aeternity/issues/3937)

File: apps/aecore/test/aec_parent_chain_cache_tests.erl [hunk 3]
```diff
@@ -44,7 +83,7 @@ cache_all_above_child_height() ->
         fun(CacheMaxSize, StartHeight, ChildTop0) ->
             meck:expect(aec_chain, top_height, fun() -> ChildTop0 end),
             {ok, _CachePid} = start_cache(StartHeight, CacheMaxSize, _Confirmations = 1),
-            timer:sleep(10),
+            timer:sleep(20),
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aecore/test/aec_parent_chain_cache_tests.erl:86]: not only initialisation but also population of the cache as well...

File: apps/aecore/src/aec_consensus_hc.erl [hunk 4]
```diff
@@ -108,7 +124,7 @@ start(Config) ->
     lager:debug("Stakers: ~p", [StakersMap]),
```
[the remark this answers is not in the document]
Comment (velzevur) [reply apps/aecore/src/aec_consensus_hc.erl:124]: This should be addressed by the #3937. I left a comment there.

**Review by uwiger — APPROVED** [review]

**Review by fabiankrol — APPROVED** [review]

**Pull request merged**